src/v/config/configuration.cc

```diff
@@ -193,6 +193,25 @@ configuration::configuration()
       2_MiB,
       // WebAssembly uses 64KiB pages and has a 32bit address space
       {.min = 64_KiB, .max = 4_GiB})
+  , data_transforms_runtime_limit_ms(
+      *this,
+      "data_transforms_runtime_limit_ms",
+      "The maximum amount of runtime for startup time of a data transform, and "
+      "for the time it takes for a single record to be transformed. Note this "
+      "is a soft limit.",
```

Comment:
>this is a soft limit

nit: might want to flesh this out a tiny bit. sample size of one, but I'm left wondering.

Reply:
I just dropped it. I want to externally try and give people an understandable limit, but we're really just guessing because we actually can only enforce the number of instructions executed in the VM. This limit will actually vary wildly based on the CPU it runs on. It's also CPU time not wall clock time. Anyways, it's a tunable (so not intended to be user visible per say), and I just expect people to jack it up if they hit problems.

Reply:
Makes sense. FWIW I agree that it's not worth describing that reasoning in the config description - tunable config w/ a generous default seems totally reasonable to me.